packages/webdriverio/src/utils/interception/devtools.js

```diff
@@ -76,22 +75,27 @@ export default class DevtoolsInterception extends Interception {
                         newBody = await overwrite(request, client)
                     }
 
+                    const isBodyUndefined = typeof newBody === 'undefined'
+                    if (isBodyUndefined) {
+                        newBody = ''
+                    }
+
                     if (typeof newBody !== 'string') {
                         newBody = JSON.stringify(newBody)
                     }
 
                     let responseCode = params.statusCode || responseStatusCode
                     let responseHeaders = [
                         ...event.responseHeaders,
-                        ...Object.entries(params.headers || {}).map(([key, value]) => { key, value })
+                        ...Object.entries(params.headers || {}).map(([name, value]) => ({ name, value }))
                     ]
 
                     /**
                      * check if local file and load it
                      */
                     const responseFilePath = path.isAbsolute(newBody) ? newBody : path.join(process.cwd(), newBody)
-                    if (fs.existsSync(responseFilePath) && canAccess(responseFilePath)) {
```

Comment:
Why do we need `fs-extra` as extra dependency if `fs.existsSync(responseFilePath)` would do the same thing?

Reply:
`fs-extra` exists anyway as a dependency of some packages.
`existsSync` - is sync method, while `pathExists` is async method.

Reply:
> existsSync - is sync method, while pathExists is async method.

What is the advantage?

Reply:
having one of sync methods here causing some issues to mocking
- pages loads slower if there are > 1000 requests
- I'm getting errors regarding invalid InterceptionId

Reply:
I would say it is simpler to promisify the `fs` methods here, this would also not require changes in the unit tests.